client/src/main/scala/org/apache/celeborn/client/commit/CommitHandler.scala

```diff
@@ -0,0 +1,365 @@
+import org.apache.celeborn.common.protocol.message.StatusCode
+import org.apache.celeborn.common.rpc.RpcEndpointRef
+import org.apache.celeborn.common.util.{ThreadUtils, Utils}
+// Can Remove this if celeborn don't support scala211 in future
+import org.apache.celeborn.common.util.FunctionConverter._
+
+case class ParallelCommitResult(
+    masterPartitionLocationMap: ConcurrentHashMap[String, PartitionLocation],
+    slavePartitionLocationMap: ConcurrentHashMap[String, PartitionLocation],
+    commitFilesFailedWorkers: ShuffleFailedWorkers)
+
+abstract class CommitHandler(
+    appId: String,
+    conf: CelebornConf,
+    allocatedWorkers: ShuffleAllocatedWorkers,
+    reducerFileGroupsMap: ShuffleFileGroups,
+    committedPartitionInfo: CommittedPartitionInfo) extends Logging {
+
+  private val pushReplicateEnabled = conf.pushReplicateEnabled
+  private val testRetryCommitFiles = conf.testRetryCommitFiles
+  private val commitEpoch = new AtomicLong()
+  private val totalWritten = new LongAdder
+  private val fileCount = new LongAdder
+
+  def getPartitionType(): PartitionType
+
+  def isStageEnd(shuffleId: Int): Boolean = false
+
+  def isStageEndOrInProcess(shuffleId: Int): Boolean = false
+
+  def isStageDataLost(shuffleId: Int): Boolean = false
+
+  def setStageEnd(shuffleId: Int): Unit
+
+  def isPartitionInProcess(shuffleId: Int, partitionId: Int): Boolean = false
+
+  def finalCommit(
+      shuffleId: Int,
+      recordWorkerFailure: ShuffleFailedWorkers => Unit): Unit
+
+  def finalPartitionCommit(
+      shuffleId: Int,
+      partitionId: Int,
+      recordWorkerFailure: ShuffleFailedWorkers => Unit): Boolean
+
+  def removeExpiredShuffle(shuffleId: Int): Unit
+
+  def getShuffleMapperAttempts(shuffleId: Int): Array[Int]
+
+  def parallelCommitFiles(
+      shuffleId: Int,
+      allocatedWorkers: util.Map[WorkerInfo, PartitionLocationInfo],
+      partitionIdOpt: Option[Int] = None): ParallelCommitResult = {
+    val shuffleCommittedInfo = committedPartitionInfo.get(shuffleId)
+    val masterPartMap = new ConcurrentHashMap[String, PartitionLocation]
+    val slavePartMap = new ConcurrentHashMap[String, PartitionLocation]
+    val commitFilesFailedWorkers = new ShuffleFailedWorkers()
+    val commitFileStartTime = System.nanoTime()
+    val parallelism = Math.min(allocatedWorkers.size(), conf.rpcMaxParallelism)
+    ThreadUtils.parmap(
+      allocatedWorkers.asScala.to,
+      "CommitFiles",
+      parallelism) { case (worker, partitionLocationInfo) =>
+      if (partitionLocationInfo.containsShuffle(shuffleId.toString)) {
+        val masterParts =
+          partitionLocationInfo.getMasterLocations(shuffleId.toString, partitionIdOpt)
+        val slaveParts = partitionLocationInfo.getSlaveLocations(shuffleId.toString, partitionIdOpt)
+        masterParts.asScala.foreach { p =>
+          val partition = new PartitionLocation(p)
+          partition.setFetchPort(worker.fetchPort)
```

Comment:
Do we need to setFetchPort here? Maybe we should check wether fetchPort is already set and possibly remove this line. We can do this later.

Reply:
It can be refactor later.